```
error[E0063]: missing field `frame_id` in initializer of `device::GpuFrameProfile<_>`
   --> /home/servo/.cargo/git/checkouts/webrender-c3596abe1cf4f320/9496efa/webrender/src/device.rs:522:9
    |
522 |         GpuFrameProfile {
    |         ^^^^^^^^^^^^^^^ missing `frame_id`
```

--- a/components/compositing/compositor.rs
+++ b/components/compositing/compositor.rs
@@ -12,7 +12,7 @@ use euclid::point::TypedPoint2D;
 use euclid::rect::TypedRect;
 use euclid::scale_factor::ScaleFactor;
 use euclid::size::TypedSize2D;
-use gfx_traits::{Epoch, FragmentType, ScrollRootId};
+use gfx_traits::{Epoch, ScrollRootId};
 use gleam::gl;
 use gleam::gl::types::{GLint, GLsizei};
 use image::{DynamicImage, ImageFormat, RgbImage};
@@ -80,22 +80,6 @@ trait ConvertScrollRootIdFromWebRender {
 impl ConvertScrollRootIdFromWebRender for webrender_traits::ServoScrollRootId {
     fn from_webrender(&self) -> ScrollRootId {
         ScrollRootId(self.0)
-    }
-}
-
-trait ConvertFragmentTypeFromWebRender {
-    fn from_webrender(&self) -> FragmentType;
-}
-
-impl ConvertFragmentTypeFromWebRender for webrender_traits::FragmentType {
-    fn from_webrender(&self) -> FragmentType {
-        match *self {
-            webrender_traits::FragmentType::FragmentBody => FragmentType::FragmentBody,
-            webrender_traits::FragmentType::BeforePseudoContent => {
-                FragmentType::BeforePseudoContent
-            }
-            webrender_traits::FragmentType::AfterPseudoContent => FragmentType::AfterPseudoContent,
-        }
     }
 }
 

--- a/components/layout/webrender_helpers.rs
+++ b/components/layout/webrender_helpers.rs
@@ -11,7 +11,7 @@ use app_units::Au;
 use euclid::{Point2D, Rect, SideOffsets2D, Size2D};
 use gfx::display_list::{BorderDetails, BorderRadii, BoxShadowClipMode, ClippingRegion};
 use gfx::display_list::{DisplayItem, DisplayList, DisplayListTraversal, StackingContextType};
-use gfx_traits::{FragmentType, ScrollRootId};
+use gfx_traits::ScrollRootId;
 use msg::constellation_msg::PipelineId;
 use style::computed_values::{image_rendering, mix_blend_mode};
 use style::computed_values::filter::{self, Filter};
@@ -419,21 +419,5 @@ trait WebRenderScrollRootIdConverter {
 impl WebRenderScrollRootIdConverter for ScrollRootId {
     fn convert_to_webrender(&self) -> webrender_traits::ServoScrollRootId {
         webrender_traits::ServoScrollRootId(self.0)
-    }
-}
-
-trait WebRenderFragmentTypeConverter {
-    fn convert_to_webrender(&self) -> webrender_traits::FragmentType;
-}
-
-impl WebRenderFragmentTypeConverter for FragmentType {
-    fn convert_to_webrender(&self) -> webrender_traits::FragmentType {
-        match *self {
-            FragmentType::FragmentBody => webrender_traits::FragmentType::FragmentBody,
-            FragmentType::BeforePseudoContent => {
-                webrender_traits::FragmentType::BeforePseudoContent
-            }
-            FragmentType::AfterPseudoContent => webrender_traits::FragmentType::AfterPseudoContent,
-        }
     }
 }

--- a/components/servo/lib.rs
+++ b/components/servo/lib.rs
@@ -99,6 +99,7 @@ use std::cmp::max;
 use std::path::PathBuf;
 use std::rc::Rc;
 use std::sync::mpsc::Sender;
+use webrender::renderer::RendererKind;
 use webvr::{WebVRThread, WebVRCompositorHandler};
 
 pub use gleam::gl;
@@ -159,9 +160,9 @@ impl<Window> Browser<Window> where Window: WindowMethods + 'static {
             };
 
             let renderer_kind = if opts::get().should_use_osmesa() {
-                webrender_traits::RendererKind::OSMesa
+                RendererKind::OSMesa
             } else {
-                webrender_traits::RendererKind::Native
+                RendererKind::Native
             };
 
             let recorder = if opts.webrender_record {

